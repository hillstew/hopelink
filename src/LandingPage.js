import React, { Component } from "react";

export default class LandingPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: ""
    };
  }

  handleChange = event => {
    this.setState({
      userName: event.target.value
    });
  };

  updateUserName = () => {
    this.props.updateUserName(this.state.userName);
    this.setState({ userName: "" });
  };

  render() {
    return (
      <div>
        <p>Please enter your name</p>
        <input
          onChange={this.handleChange}
          value={this.state.userName}
          type="text"
        />
        <button onClick={this.updateUserName}>Save</button>
      </div>
    );
  }
}
