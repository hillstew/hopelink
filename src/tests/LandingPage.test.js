import React from "react";
import { shallow } from "enzyme";
import LandingPage from "../LandingPage";

let updateUserNameMock = jest.fn();

describe("Landing Page", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LandingPage updateUserName={updateUserNameMock} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should updateState onChange of the input", () => {
    const mockEvent = { target: { value: "Hillary" } };
    wrapper.instance().updateState(mockEvent);
    expect(wrapper.state("userName").length).toEqual(7);
  });

  it("should updateUserName when the save button is clicked", () => {
    const spy = jest.spyOn(wrapper.instance(), 'updateUserName')
    wrapper.instance().forceUpdate()
    wrapper.find("button").simulate("click");
    expect(spy).toHaveBeenCalled();
  });
});