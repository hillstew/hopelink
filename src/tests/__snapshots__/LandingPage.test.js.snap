// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page should match snapshot 1`] = `
<div
  className="landing-page-div"
>
  <p>
    Please enter your name
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Save
  </button>
</div>
`;
